src/component/presentation/preview.tsx

```diff
@@ -25,22 +27,79 @@ class PatternWrapper extends React.Component<{}, PatternWrapperState> {
 	}
 }
 
+export interface HighlightAreaProps extends ClientRect {
+	opacity?: number;
+}
+
 export interface PreviewProps {
 	page?: Page;
+	selectedElementId?: number[];
+}
+
+export interface PreviewState {
+	highlightArea: HighlightAreaProps;
 }
 
 @observer
-export class Preview extends React.Component<PreviewProps> {
+export class Preview extends React.Component<PreviewProps, PreviewState> {
 	private patternFactories: { [folder: string]: React.StatelessComponent };
 
 	public constructor(props: PreviewProps) {
 		super(props);
 		this.patternFactories = {};
+
+		this.state = {
+			highlightArea: {
+				bottom: 0,
+				height: 0,
+				left: 0,
+				opacity: 0,
+				right: 0,
+				top: 0,
+				width: 0
+			}
+		};
+	}
+
+	public componentDidUpdate(prevProps: PreviewProps, prevState: PreviewState): void {
+		if (this.state.highlightArea.opacity !== 0) {
+			setTimeout(() => {
+				this.setState({
+					highlightArea: {
+						...this.state.highlightArea,
+						opacity: 0
+					}
+				});
+			}, 500);
+		}
 	}
 
 	public render(): JSX.Element | null {
+		SmoothscrollPolyfill.polyfill();
```

Comment:
the polyfill should be executed somewhere else i think. Perhaps the PreviewApps constructor would be a more fitting place for now?